### PR TITLE
build(aio): improve accessor rendering

### PR DESCRIPTION
**Pull request by petebacondarwin**
Includes an update to `dgeni-packages@0.22.0` which provides more info
about accessors if required.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: aio/package.json [hunk 1]
```diff
@@ -99,7 +99,7 @@
     "concurrently": "^3.4.0",
     "cross-spawn": "^5.1.0",
     "dgeni": "^0.4.7",
-    "dgeni-packages": "^0.21.4",
+    "dgeni-packages": "0.22.0",
```
Comment (gkalpak) [inline aio/package.json]: Is the drop of `^` intentional?

File: aio/tools/transforms/templates/api/lib/memberHelpers.html [hunk 2]
```diff
@@ -18,8 +18,8 @@
 
 {%- macro renderMember(member, truncateLines) -%}
     {%- if member.accessibility !== 'public' %}{$ member.accessibility $} {% endif -%}
-    {%- if member.isGetAccessor or member.isReadonly %}get {% endif -%}
-    {%- if member.isSetAccessor %}set {% endif -%}
+    {%- if (member.isGetAccessor or member.isReadonly) and not member.isSetAccessor %}get {% endif -%}
+    {%- if member.isSetAccessor and not member.isGetAccessor %}set {% endif -%}
```
Comment (gkalpak) [inline aio/tools/transforms/templates/api/lib/memberHelpers.html]: Can a `member` be both a get and a set accessor?

File: aio/package.json (hunk 1 above)
Comment (petebacondarwin) [reply aio/package.json]: Yes, I decided it was safer not to have a range here - that is partially what caused the breakage to @Foxandxss the other day. Due to our lockfile usage we would manually update to 0.22.1, etc anyway.

File: aio/tools/transforms/templates/api/lib/memberHelpers.html (hunk 2 above)
Comment (petebacondarwin) [reply aio/tools/transforms/templates/api/lib/memberHelpers.html]: Yes, a property member can be both, in which case it also has properties: `getAccessor` and `setAccessor` which contain the information and docs for those accessors. This is similar to how function overloads are handled.
The type and description of the property member is computed from the get and set accessors but we could access and render information about each of the accessors if we wished. At this stage it doesn't seem important.

Comment (petebacondarwin) [reply aio/tools/transforms/templates/api/lib/memberHelpers.html]: See `NgForOf` (https://angular.io/api/common/NgForOf#ngForTrackBy) for an example of how it looks right now. I decided that if a property has both a getter and a setter then, from the point of view of the application developer, it is just a property.

File: aio/package.json (hunk 1 above)
Comment (gkalpak) [reply aio/package.json]: By introducing the `--freeze-lockfile` flag we shouldn't run into the @Foxandxss incident again, but fine to drop the `^` anyway :smiley:

**Review by gkalpak — APPROVED** [review]

**Pull request closed without merging**